model/location_credential.rb

```diff
@@ -3,27 +3,45 @@
 require_relative "../model"
 require "aws-sdk-ec2"
 require "aws-sdk-iam"
+# :nocov:
+require "aws-sdk-sts" if Config.test? ? ENV["CLOVER_FREEZE"] != "1" : Config.aws_postgres_iam_access
+# :nocov:
 
 class LocationCredential < Sequel::Model
   plugin ResourceMethods, encrypted_columns: [:access_key, :secret_key]
   many_to_one :project
   many_to_one :location, key: :id
 
+  def credentials
+    @credentials ||= if assume_role
+      Aws::AssumeRoleCredentials.new(role_arn: assume_role, role_session_name: Config.aws_role_session_name)
+    else
+      Aws::Credentials.new(access_key, secret_key)
+    end
+  end
+
   def client
-    Aws::EC2::Client.new(access_key_id: access_key, secret_access_key: secret_key, region: location.name)
+    @client ||= Aws::EC2::Client.new(region: location.name, credentials:)
```

Comment:
@furkansahin I also added caching client here to be consistent with your iam_client feedback, but wanted to point this out in case there's a good reason this wasn't cached previously